openeo/testing/results.py

```diff
@@ -88,6 +91,129 @@ def _as_xarray_dataarray(data: Union[str, Path, xarray.DataArray]) -> xarray.Dat
     return data
 
 
+def ascii_art(diff_data: DataArray) -> str:
+    scale: int = max(1, int(diff_data.sizes["x"] / 100))
+    data_max = diff_data.max().item()
+    if data_max == 0:
+        data_max = 1
+    grayscale_characters = "$@B%8&WM#*oahkbdpqwmZO0QLCJUYXzcvunxrjft/\|()1{}[]?-_+~<>i!lI;:,\"^`'. "
```

Comment:
just curious: where does this "gradient" come from?

also note that with the (github.com) font I'm currently looking looking at this, it doesn't seem to be monotonically increasing in "lightness":  e.g. `+` comes after `_`  but it looks "darker"

Reply:
I also noticed this warning when running tests:
```
  .../openeo-python-client/openeo/testing/results.py:99: SyntaxWarning: invalid escape sequence '\|'
    grayscale_characters = "$@B%8&WM#*oahkbdpqwmZO0QLCJUYXzcvunxrjft/\|()1{}[]?-_+~<>i!lI;:,\"^`'. "
```